synopsys-task/src/synopsys-task/tools-parameter.ts

```diff
@@ -243,39 +258,49 @@ export class SynopsysToolsParameter {
 
     // Check and put environment variable for fix pull request
     if (parseToBoolean(inputs.BLACKDUCK_FIXPR_ENABLED)) {
-      console.log("Black Duck Fix PR is enabled");
-      blackduckData.data.blackduck.fixpr = this.setBlackDuckFixPrInputs();
-      blackduckData.data.azure = await this.getAzureRepoInfo();
-    } else {
-      // Disable fix pull request for adapters
-      blackduckData.data.blackduck.fixpr = { enabled: false };
+      if (isPullRequest()) {
+        taskLib.warning(
+          "Black Duck Fix PR is enabled but it will be ignored since it is a PR/MR scan"
+        );
+      } else {
+        console.log("Black Duck Fix PR is enabled");
+        blackduckData.data.blackduck.fixpr = this.setBlackDuckFixPrInputs();
+        blackduckData.data.azure = await this.getAzureRepoInfo();
+      }
     }
 
     if (parseToBoolean(inputs.BLACKDUCK_AUTOMATION_PRCOMMENT)) {
-      console.info("BlackDuck Automation comment is enabled");
-      blackduckData.data.azure = await this.getAzureRepoInfo();
-      blackduckData.data.environment = this.setEnvironmentScanPullData();
-      blackduckData.data.blackduck.automation.prcomment = true;
-      blackduckData.data;
+      if (!isPullRequest()) {
+        taskLib.warning(
+          "Black Duck PR comment is enabled but it will be ignored since it is not a PR/MR scan"
+        );
+      } else {
+        console.info("BlackDuck PR comment is enabled");
+        blackduckData.data.azure = await this.getAzureRepoInfo();
+        blackduckData.data.environment = this.setEnvironmentScanPullData();
+        blackduckData.data.blackduck.automation = { prcomment: true };
+        blackduckData.data;
+      }
     }
 
-    const buildReason =
-      taskLib.getVariable(AZURE_ENVIRONMENT_VARIABLES.AZURE_BUILD_REASON) || "";
+    if (parseToBoolean(inputs.ENABLE_NETWORK_AIRGAP)) {
+      blackduckData.data.network = { airGap: true };
+    }
 
-    if (
-      parseToBoolean(inputs.BLACKDUCK_REPORTS_SARIF_CREATE) ||
-      parseToBoolean(inputs.BLACKDUCK_REPORTS_SARIF_CREATE_CLASSIC_EDITOR)
-    ) {
-      if (buildReason !== AZURE_BUILD_REASON.PULL_REQUEST) {
+    if (parseToBoolean(inputs.BLACKDUCK_REPORTS_SARIF_CREATE)) {
+      if (!isPullRequest()) {
```

Comment:
can we store the isPullRequest() value in some variable and reuse here?

Reply:
updated